cli/catalog-api-v1/src/error.rs

```diff
@@ -14,5 +13,28 @@ pub enum MessageType {
     #[serde(rename = "constraints_too_tight")]
     ConstraintsTooTight,
 
+    #[serde(untagged)]
```

Comment:
Its a bit of an unusual usage of `untagged` to me at least.
Why did you change this to the variant?

Reply:
I don't exactly understand the question, but I wanted `"constraints_too_tight"` to deserialize to `ConstraintsTooTight` not `Unknown("constraints_too_tight")`

I found this usage via https://github.com/serde-rs/serde/issues/912#issuecomment-1868785603

Reply:
oh what, it did deserialize as Unknown even though there's a variant for it 🤔
that's surprising, i guess that alone is a reason to keep the tests lol

nevermind then, untagged on the field seems to be the right use then